pymc3/distributions/timeseries.py

```diff
@@ -71,3 +71,29 @@ def logp(self, x):
 
         innov_like = Normal.dist(mu=x_im1 + mu, tau=tau, sd=sd).logp(x_i)
         return init.logp(x[0]) + T.sum(innov_like)
+
+
+
+class GARCH(Continuous):
+    def __init__(self, alpha_0=None, alpha_1=None, beta_1=None, sigma_0=None, *args, **kwargs):
+        super(GARCH, self).__init__(*args, **kwargs)
+
+        self.alpha_0 = alpha_0
+        self.alpha_1 = alpha_1
+        self.beta_1 = beta_1
+        self.sigma_0 = sigma_0
+        self.mean = 0
+
+    def logp(self, x):
+        sigma = self.sigma_0
+        alpha_0 = self.alpha_0
+        alpha_1 = self.alpha_1
+        beta_1 = self.beta_1
+
+        x_im1 = x[:-1]
+        x_i = x[1:]
+        sigma = T.sqrt(alpha_0 + alpha_1 * (x_im1/sigma)**2
+                            + beta_1 * sigma**2)
+        innov_like = Normal.dist(0., beta_1=beta_1, sd=sigma).logp(x_i)
```

Comment:
I have another bit of code to send in. Will do that.
On 9 Feb 2016 23:14, "Chris Fonnesbeck" <notifications@github.com> wrote:

> In pymc3/distributions/timeseries.py
> <https://github.com/pymc-devs/pymc3/pull/967#discussion_r52383014>:
>
> > +        self.alpha_1 = alpha_1
> > +        self.beta_1 = beta_1
> > +        self.sigma_0 = sigma_0
> > +        self.mean = 0
> > +
> > +    def logp(self, x):
> > +        sigma = self.sigma_0
> > +        alpha_0 = self.alpha_0
> > +        alpha_1 = self.alpha_1
> > +        beta_1 = self.beta_1
> > +
> > +        x_im1 = x[:-1]
> > +        x_i = x[1:]
> > +        sigma = T.sqrt(alpha_0 + alpha_1 * (x_im1/sigma)**2
> > +                            + beta_1 * sigma**2)
> > +        innov_like = Normal.dist(0., beta_1=beta_1, sd=sigma).logp(x_i)
>
> There are still too many arguments. Should only be two.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pymc-devs/pymc3/pull/967/files#r52383014>.
>